corehq/s3/s3.py

```diff
@@ -0,0 +1,48 @@
+from django.conf import settings
+
+import boto3
+from botocore.utils import fix_s3_host
+from botocore.exceptions import ClientError
+
+
+class ObjectStore(object):
+
+    def __init__(self, domain=None):
+        self.domain = domain
+        self.resource = boto3.resource(
+            's3',
+            use_ssl=settings.RIAKCS_SSL,
+            endpoint_url='http://{}:{}'.format(settings.RIAKCS_HOST, settings.RIAKCS_PORT),
+            aws_access_key_id=settings.RIAKCS_KEY,
+            aws_secret_access_key=settings.RIAKCS_SECRET,
+        )
+
+        # boto3 automatically overwrites the endpoint url: https://github.com/boto/boto3/issues/259
+        self.resource.meta.client.meta.events.unregister('before-sign.s3', fix_s3_host)
+        self.bucket = self._init_bucket()
```

Comment:
not a fan of doing work in __init__ functions. Would be better to make this lazily executed with a memoized property